vault/token_store.go

```diff
@@ -3295,14 +3295,6 @@ func (ts *TokenStore) handleRevokeOrphan(ctx context.Context, req *logical.Reque
 		return logical.ErrorResponse("missing token ID"), logical.ErrInvalidRequest
 	}
 
-	// Check if the client token has sudo/root privileges for the requested path
-	isSudo := ts.System().(extendedSystemView).SudoPrivilege(ctx, req.MountPoint+req.Path, req.ClientToken)
-
-	if !isSudo {
-		return logical.ErrorResponse("root or sudo privileges required to revoke and orphan"),
-			logical.ErrInvalidRequest
-	}
-
```

Comment:
Note to reviewer: This code is now redundant, as sudo requirements are, with this change, enforced by the standard framework used for all other endpoints, and custom code is not required.

This change also means that the HTTP error code for not having sudo rights and trying to use this endpoint, will change from the technically incorrect 400 (bad request) to more correct 403 (forbidden), consistent with how other endpoints behave.

Reply:
I'm not very familiar where this would be checked in the framework. Perhaps @hashicorp/vault-core could help confirm this.

Reply:
It will be checked via the following flow:

1. The amended PathsSpecial.Root earlier in this file's diff now matches the request
2. `(*Router).Mount` will capture the defined root paths into a `radix.Tree`: https://github.com/hashicorp/vault/blob/0b0b15f968bb243b29544d6a7f2652137e07d632/vault/router.go#L198
3. `(*Core).CheckToken` will call `(*Router).RootPath` to query that `radix.Tree` during request handling: https://github.com/hashicorp/vault/blob/0b0b15f968bb243b29544d6a7f2652137e07d632/vault/request_handling.go#L302
4. and then inform `(*Core).performPolicyChecks` that it needs to look for a root token or sudo capability during standard ACL policy evaluation: https://github.com/hashicorp/vault/blob/0b0b15f968bb243b29544d6a7f2652137e07d632/vault/request_handling.go#L397-L400

Reply:
Hi @maxb! Would it be possible to split this sudo check out into its own PR? The @hashicorp/vault-core team can review it separately and we'll be able to merge this PR :)

Reply:
> Alternatively, I guess I could split just removing the single block of code that this thread is attached to into a separate PR. That would amount to making the code redundant by implementing the more normal check, but deferring removal of the redundant code to a separate change.

I think that's the more preferable path. If I understand correctly, if we keep the sudo check in this PR, the function will still work as before and will return an error when a request is made without root privileges (even though this check is redundant, it shouldn't break anything).

Reply:
Thanks!